Fixed issue #1410

dplefevre opened the pull request:
This addresses issue #1410 "trim_user" and "exclude_replies" missing from user_timeline docstring.

Reviewed and tested these options, and updated api.rst and parameters.rst

Harmon758 reviewed (changes requested):
This PR only updates the documentation and doesn't actually add the allowed parameters that the documentation would reflect.

Also the erroneous documentation for the `page` parameter needs to be removed, as it was never an allowed parameter for this method.

Additionally, `include_entities` should be added as an allowed parameter, with corresponding documentation as well, as it's also missing.

It might also be worth reordering `count` to be the first allowed parameter for consistency with the API's documentation and for simpler usage of the method.

dplefevre commented on docs/parameters.rst:
```diff
@@ -4,6 +4,7 @@
 .. |cursor| replace:: Breaks the results into pages. Provide a value of  -1 to begin paging. Provide values as returned to in the response body's next_cursor and previous_cursor attributes to page back and forth in the list.
 .. |date| replace:: Permits specifying a start date for the report. The date should be formatted YYYY-MM-DD.
 .. |exclude| replace:: Setting this equal to hashtags will remove all hashtags from the trends list.
+.. |exclude_replies| replace:: A boolean which will prevent reply tweets from being included with the returned timeline. Note that the API will return a number of tweets equal to the count parameter before replies are filtered out. Defaults to True.
```
I moved the description over to api.rst for now, and removed this line from parameters.rst.

dplefevre commented:
Thank you for the feedback. I believe I have addressed all of the issues you raised in my latest push. Please have a look and let me know if you need any additional changes. Full disclosure, this is my first attempt at an open source contribution, so I hope you do not mind being a bit patient with me.

Harmon758 commented on docs/api.rst:
```diff
@@ -47,16 +47,23 @@ This page contains some basic documentation for the Tweepy module.
-   :param count: |count|
-   :param page: |page|
+   :param trim_user: |trim_user|
+   :param exclude_replies: A boolean which will prevent reply tweets from
+                           being included with the returned timeline. Note
+                           that the API will return a number of tweets equal
+                           to the count parameter before replies are filtered
+                           out. Defaults to True.
```
From my testing, this seems to default to False.
In any case, I think just using the description of the parameter as is from Twitter's documentation of the endpoint should suffice:
```suggestion
   :param exclude_replies: This parameter will prevent replies from appearing
                           in the returned timeline. Using ``exclude_replies``
                           with the ``count`` parameter will mean you will
                           receive up-to count Tweets — this is because the
                           ``count`` parameter retrieves that many Tweets
                           before filtering out retweets and replies.
```

Harmon758 reviewed (changes requested):
> Full disclosure, this is my first attempt at an open source contribution, so I hope you do not mind being a bit patient with me.

No problem, that's what reviews are for.

Harmon758 commented:
Thanks for the contribution!
